test/model/message_test.dart

```diff
@@ -576,4 +603,197 @@ void main() {
         .reactions.isNotNull().jsonEquals([eg.unicodeEmojiReaction]);
     });
   });
+
+  group('handleSubmessageEvent', () {
+    late StreamMessage streamMessage;
+
+    Future<int> preparePollMessage({List<PollOption>? options}) async {
+      List<PollOption> effectiveOptions = options
+        ?? [PollOption(text: 'foo'), PollOption(text: 'bar')];
+      const messageId = 123;
+      streamMessage = eg.streamMessage(
+        id: messageId,
+        sender: eg.selfUser,
+      );
+      final submessages = [
+        eg.submessage(
+          senderId: eg.selfUser.userId,
+          content: jsonEncode(PollWidgetData(extraData: PollWidgetExtraData(
+            question: 'Favorite word',
+            options: effectiveOptions.map((e) => e.text).toList(),
+          )),
+        )),
+        for (int i = 0; i < effectiveOptions.length; i++)
+          ...[
+            for (final voter in effectiveOptions[i].voters)
+            eg.submessage(
+              senderId: voter,
+              content: jsonEncode(PollVoteEventSubmessage(
+                key: PollEventSubmessage.optionKey(senderId: null, idx: i),
+                op: PollVoteOp.add,
+              )),
+            )
+          ]
+      ];
+      await prepare();
+      await prepareMessageWithSubmessage(streamMessage, submessages);
+      return messageId;
+    }
+
+    test('message is unknown', () async {
+      await prepare();
+      await store.handleEvent(
+        eg.submessageEvent(1000, eg.otherUser.userId, content: '{}')
+      );
+      checkNotNotified();
+    });
+
+    test('message has no submessages', () async {
+      final streamMessage = eg.streamMessage(id: 123);
```

Comment:
nit:
```suggestion
      final message = eg.streamMessage(id: 123);
```

The fact it's a stream message isn't important, right? So can leave that out of the story this tells to the reader. (It's still there in the `eg.streamMessage` call, but gets less emphasis.)